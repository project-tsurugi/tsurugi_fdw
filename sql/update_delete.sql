/* DDL */
---int
CREATE TABLE update_delete_int1(col1 int PRIMARY KEY, col2 int) TABLESPACE tsurugi;
CREATE FOREIGN TABLE update_delete_int1(col1 int, col2 int) SERVER tsurugidb;

---bigint
CREATE TABLE update_delete_bigint1(col1 int PRIMARY KEY, col2 bigint) TABLESPACE tsurugi;
CREATE FOREIGN TABLE update_delete_bigint1(col1 int, col2 bigint) SERVER tsurugidb;

---real
CREATE TABLE update_delete_real1(col1 int PRIMARY KEY, col2 real) TABLESPACE tsurugi;
CREATE FOREIGN TABLE update_delete_real1(col1 int, col2 real) SERVER tsurugidb;

---double precision
CREATE TABLE update_delete_double1(col1 int PRIMARY KEY, col2 double precision) TABLESPACE tsurugi;
CREATE FOREIGN TABLE update_delete_double1(col1 int, col2 double precision) SERVER tsurugidb;

---char
CREATE TABLE update_delete_char1(col1 int PRIMARY KEY, col2 char(2)) TABLESPACE tsurugi;
CREATE FOREIGN TABLE update_delete_char1(col1 int, col2 char(2)) SERVER tsurugidb;

---varchar
CREATE TABLE update_delete_varchar1(col1 int PRIMARY KEY, col2 varchar(2)) TABLESPACE tsurugi;
CREATE FOREIGN TABLE update_delete_varchar1(col1 int, col2 varchar(2)) SERVER tsurugidb;

/* DML */
---int
SELECT * FROM update_delete_int1;
INSERT INTO update_delete_int1 (col1, col2) VALUES (1, 1);
INSERT INTO update_delete_int1 (col1, col2) VALUES (2, 2);
SELECT * FROM update_delete_int1;

UPDATE update_delete_int1 SET col2 = col2+2147483645;
SELECT * FROM update_delete_int1;

DELETE FROM update_delete_int1 WHERE col2 = 2147483647;
SELECT * FROM update_delete_int1;

DELETE FROM update_delete_int1 WHERE col3 = 1;
SELECT * FROM update_delete_int1;

---bigint
SELECT * FROM update_delete_bigint1;
INSERT INTO update_delete_bigint1 (col1, col2) VALUES (1, 1);
INSERT INTO update_delete_bigint1 (col1, col2) VALUES (2, 2);
SELECT * FROM update_delete_bigint1;

UPDATE update_delete_bigint1 SET col2 = col2+9223372036854775805;
SELECT * FROM update_delete_bigint1;

DELETE FROM update_delete_bigint1 WHERE col2 = 9223372036854775807;
SELECT * FROM update_delete_bigint1;

DELETE FROM update_delete_bigint1 WHERE col3 = 1;
SELECT * FROM update_delete_bigint1;

---real
SELECT * FROM update_delete_real1;
INSERT INTO update_delete_real1 (col1, col2) VALUES (1, 1.1);
INSERT INTO update_delete_real1 (col1, col2) VALUES (2, 0.1);
INSERT INTO update_delete_real1 (col1, col2) VALUES (3, 3.4);
SELECT * FROM update_delete_real1;

UPDATE update_delete_real1 SET col2 = col2+1;
SELECT * FROM update_delete_real1;  -- fix Tsurugi1.0.0-BETA4

DELETE FROM update_delete_real1 WHERE col2 = cast(4.4 as real);  -- see tsurugi-issues#736
SELECT * FROM update_delete_real1;  -- fix Tsurugi1.0.0-BETA4

DELETE FROM update_delete_real1 WHERE col3 = 0.1;
SELECT * FROM update_delete_real1;  -- fix Tsurugi1.0.0-BETA4

---double precision
SELECT * FROM update_delete_double1;
INSERT INTO update_delete_double1 (col1, col2) VALUES (1, 1.1);
INSERT INTO update_delete_double1 (col1, col2) VALUES (2, 0.1);
INSERT INTO update_delete_double1 (col1, col2) VALUES (3, 3.4);
SELECT * FROM update_delete_double1;

UPDATE update_delete_double1 SET col2 = col2+1;
SELECT * FROM update_delete_double1;

DELETE FROM update_delete_double1 WHERE col2 = 4.4;
SELECT * FROM update_delete_double1;

DELETE FROM update_delete_double1 WHERE col3 = 0.1;
SELECT * FROM update_delete_double1;

---char
SELECT * FROM update_delete_char1;
INSERT INTO update_delete_char1 (col1, col2) VALUES (1, 'ab');
INSERT INTO update_delete_char1 (col1, col2) VALUES (2, 'kl');
SELECT * FROM update_delete_char1;

UPDATE update_delete_char1 SET col2 = 'uv' WHERE col2 = 'ab';
SELECT * FROM update_delete_char1;

UPDATE update_delete_char1 SET col2 = 'uvd' WHERE col2 = 'kl';
SELECT * FROM update_delete_char1;

DELETE FROM update_delete_char1 WHERE col2 = 'kl';
SELECT * FROM update_delete_char1;

DELETE FROM update_delete_char1 WHERE col3 = 'ab';
SELECT * FROM update_delete_char1;

---varchar
SELECT * FROM update_delete_varchar1;
INSERT INTO update_delete_varchar1 (col1, col2) VALUES (1, 'ab');
INSERT INTO update_delete_varchar1 (col1, col2) VALUES (2, 'kl');
SELECT * FROM update_delete_varchar1;

UPDATE update_delete_varchar1 SET col2 = 'uv' WHERE col2 = 'ab';
SELECT * FROM update_delete_varchar1;

UPDATE update_delete_varchar1 SET col2 = 'uvd' WHERE col2 = 'kl';
SELECT * FROM update_delete_varchar1;

DELETE FROM update_delete_varchar1 WHERE col2 = 'kl';
SELECT * FROM update_delete_varchar1;

DELETE FROM update_delete_varchar1 WHERE col3 = 'ab';
SELECT * FROM update_delete_varchar1;

/* DDL */
DROP TABLE update_delete_int1;
DROP TABLE update_delete_bigint1;
DROP TABLE update_delete_real1;
DROP TABLE update_delete_double1;
DROP TABLE update_delete_char1;
DROP TABLE update_delete_varchar1;

DROP FOREIGN TABLE update_delete_int1;
DROP FOREIGN TABLE update_delete_bigint1;
DROP FOREIGN TABLE update_delete_real1;
DROP FOREIGN TABLE update_delete_double1;
DROP FOREIGN TABLE update_delete_char1;
DROP FOREIGN TABLE update_delete_varchar1;
