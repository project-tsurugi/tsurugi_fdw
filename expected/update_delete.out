/* Test setup: DDL of the Tsurugi */
SELECT tg_execute_ddl('tsurugidb', '
    CREATE TABLE update_delete_int1 (
        col1 int PRIMARY KEY,
        col2 int
    )
');
 tg_execute_ddl 
----------------
 CREATE TABLE
(1 row)

SELECT tg_execute_ddl('tsurugidb', '
    CREATE TABLE update_delete_bigint1 (
        col1 int PRIMARY KEY,
        col2 bigint
    )
');
 tg_execute_ddl 
----------------
 CREATE TABLE
(1 row)

SELECT tg_execute_ddl('tsurugidb', '
    CREATE TABLE update_delete_real1 (
        col1 int PRIMARY KEY,
        col2 real
    )
');
 tg_execute_ddl 
----------------
 CREATE TABLE
(1 row)

SELECT tg_execute_ddl('tsurugidb', '
    CREATE TABLE update_delete_double1 (
        col1 int PRIMARY KEY,
        col2 double precision
    )
');
 tg_execute_ddl 
----------------
 CREATE TABLE
(1 row)

SELECT tg_execute_ddl('tsurugidb', '
    CREATE TABLE update_delete_char1 (
        col1 int PRIMARY KEY,
        col2 char(2)
    )
');
 tg_execute_ddl 
----------------
 CREATE TABLE
(1 row)

SELECT tg_execute_ddl('tsurugidb', '
    CREATE TABLE update_delete_varchar1 (
        col1 int PRIMARY KEY,
        col2 varchar(2)
    )
');
 tg_execute_ddl 
----------------
 CREATE TABLE
(1 row)

/* Test setup: DDL of the PostgreSQL */
---int
CREATE FOREIGN TABLE update_delete_int1 (
    col1 int,
    col2 int
) SERVER tsurugidb;
---bigint
CREATE FOREIGN TABLE update_delete_bigint1 (
    col1 int,
    col2 bigint
) SERVER tsurugidb;
---real
CREATE FOREIGN TABLE update_delete_real1 (
    col1 int,
    col2 real
) SERVER tsurugidb;
---double precision
CREATE FOREIGN TABLE update_delete_double1 (
    col1 int,
    col2 double precision
) SERVER tsurugidb;
---char
CREATE FOREIGN TABLE update_delete_char1 (
    col1 int,
    col2 char(2)
) SERVER tsurugidb;
---varchar
CREATE FOREIGN TABLE update_delete_varchar1 (
    col1 int,
    col2 varchar(2)
) SERVER tsurugidb;
/* DML */
---int
SELECT * FROM update_delete_int1 ORDER BY col1;
 col1 | col2 
------+------
(0 rows)

INSERT INTO update_delete_int1 (col1, col2) VALUES (1, 1);
INSERT INTO update_delete_int1 (col1, col2) VALUES (2, 2);
SELECT * FROM update_delete_int1 ORDER BY col1;
 col1 | col2 
------+------
    1 |    1
    2 |    2
(2 rows)

UPDATE update_delete_int1 SET col2 = col2+2147483645;
SELECT * FROM update_delete_int1 ORDER BY col1;
 col1 |    col2    
------+------------
    1 | 2147483646
    2 | 2147483647
(2 rows)

DELETE FROM update_delete_int1 WHERE col2 = 2147483647;
SELECT * FROM update_delete_int1 ORDER BY col1;
 col1 |    col2    
------+------------
    1 | 2147483646
(1 row)

DELETE FROM update_delete_int1 WHERE col3 = 1;
ERROR:  column "col3" does not exist
LINE 1: DELETE FROM update_delete_int1 WHERE col3 = 1;
                                             ^
HINT:  Perhaps you meant to reference the column "update_delete_int1.col1" or the column "update_delete_int1.col2".
SELECT * FROM update_delete_int1 ORDER BY col1;
 col1 |    col2    
------+------------
    1 | 2147483646
(1 row)

---bigint
SELECT * FROM update_delete_bigint1 ORDER BY col1;
 col1 | col2 
------+------
(0 rows)

INSERT INTO update_delete_bigint1 (col1, col2) VALUES (1, 1);
INSERT INTO update_delete_bigint1 (col1, col2) VALUES (2, 2);
SELECT * FROM update_delete_bigint1 ORDER BY col1;
 col1 | col2 
------+------
    1 |    1
    2 |    2
(2 rows)

UPDATE update_delete_bigint1 SET col2 = col2+9223372036854775805;
SELECT * FROM update_delete_bigint1 ORDER BY col1;
 col1 |        col2         
------+---------------------
    1 | 9223372036854775806
    2 | 9223372036854775807
(2 rows)

DELETE FROM update_delete_bigint1 WHERE col2 = 9223372036854775807;
SELECT * FROM update_delete_bigint1 ORDER BY col1;
 col1 |        col2         
------+---------------------
    1 | 9223372036854775806
(1 row)

DELETE FROM update_delete_bigint1 WHERE col3 = 1;
ERROR:  column "col3" does not exist
LINE 1: DELETE FROM update_delete_bigint1 WHERE col3 = 1;
                                                ^
HINT:  Perhaps you meant to reference the column "update_delete_bigint1.col1" or the column "update_delete_bigint1.col2".
SELECT * FROM update_delete_bigint1 ORDER BY col1;
 col1 |        col2         
------+---------------------
    1 | 9223372036854775806
(1 row)

---real
SELECT * FROM update_delete_real1 ORDER BY col1;
 col1 | col2 
------+------
(0 rows)

INSERT INTO update_delete_real1 (col1, col2) VALUES (1, 1.1);
INSERT INTO update_delete_real1 (col1, col2) VALUES (2, 0.1);
INSERT INTO update_delete_real1 (col1, col2) VALUES (3, 3.4);
SELECT * FROM update_delete_real1 ORDER BY col1;
 col1 | col2 
------+------
    1 |  1.1
    2 |  0.1
    3 |  3.4
(3 rows)

UPDATE update_delete_real1 SET col2 = col2+1;
SELECT * FROM update_delete_real1 ORDER BY col1;  -- fix Tsurugi1.0.0-BETA4
 col1 | col2 
------+------
    1 |  2.1
    2 |  1.1
    3 |  4.4
(3 rows)

DELETE FROM update_delete_real1 WHERE col2 = cast(4.4 as real);  -- see tsurugi-issues#736
SELECT * FROM update_delete_real1 ORDER BY col1;  -- fix Tsurugi1.0.0-BETA4
 col1 | col2 
------+------
    1 |  2.1
    2 |  1.1
(2 rows)

DELETE FROM update_delete_real1 WHERE col3 = 0.1;
ERROR:  column "col3" does not exist
LINE 1: DELETE FROM update_delete_real1 WHERE col3 = 0.1;
                                              ^
HINT:  Perhaps you meant to reference the column "update_delete_real1.col1" or the column "update_delete_real1.col2".
SELECT * FROM update_delete_real1 ORDER BY col1;  -- fix Tsurugi1.0.0-BETA4
 col1 | col2 
------+------
    1 |  2.1
    2 |  1.1
(2 rows)

---double precision
SELECT * FROM update_delete_double1 ORDER BY col1;
 col1 | col2 
------+------
(0 rows)

INSERT INTO update_delete_double1 (col1, col2) VALUES (1, 1.1);
INSERT INTO update_delete_double1 (col1, col2) VALUES (2, 0.1);
INSERT INTO update_delete_double1 (col1, col2) VALUES (3, 3.4);
SELECT * FROM update_delete_double1 ORDER BY col1;
 col1 | col2 
------+------
    1 |  1.1
    2 |  0.1
    3 |  3.4
(3 rows)

UPDATE update_delete_double1 SET col2 = col2+1;
SELECT * FROM update_delete_double1 ORDER BY col1;
 col1 | col2 
------+------
    1 |  2.1
    2 |  1.1
    3 |  4.4
(3 rows)

DELETE FROM update_delete_double1 WHERE col2 = 4.4;
SELECT * FROM update_delete_double1 ORDER BY col1;
 col1 | col2 
------+------
    1 |  2.1
    2 |  1.1
(2 rows)

DELETE FROM update_delete_double1 WHERE col3 = 0.1;
ERROR:  column "col3" does not exist
LINE 1: DELETE FROM update_delete_double1 WHERE col3 = 0.1;
                                                ^
HINT:  Perhaps you meant to reference the column "update_delete_double1.col1" or the column "update_delete_double1.col2".
SELECT * FROM update_delete_double1 ORDER BY col1;
 col1 | col2 
------+------
    1 |  2.1
    2 |  1.1
(2 rows)

---char
SELECT * FROM update_delete_char1 ORDER BY col1;
 col1 | col2 
------+------
(0 rows)

INSERT INTO update_delete_char1 (col1, col2) VALUES (1, 'ab');
INSERT INTO update_delete_char1 (col1, col2) VALUES (2, 'kl');
SELECT * FROM update_delete_char1 ORDER BY col1;
 col1 | col2 
------+------
    1 | ab
    2 | kl
(2 rows)

UPDATE update_delete_char1 SET col2 = 'uv' WHERE col2 = 'ab';
SELECT * FROM update_delete_char1 ORDER BY col1;
 col1 | col2 
------+------
    1 | uv
    2 | kl
(2 rows)

UPDATE update_delete_char1 SET col2 = 'uvd' WHERE col2 = 'kl';
ERROR:  Failed to execute statement to Tsurugi. (13)
Tsurugi Server Error: VALUE_TOO_LONG_EXCEPTION (SQL-02058: evaluated value was too long)
SELECT * FROM update_delete_char1 ORDER BY col1;
 col1 | col2 
------+------
    1 | uv
    2 | kl
(2 rows)

DELETE FROM update_delete_char1 WHERE col2 = 'kl';
SELECT * FROM update_delete_char1 ORDER BY col1;
 col1 | col2 
------+------
    1 | uv
(1 row)

DELETE FROM update_delete_char1 WHERE col3 = 'ab';
ERROR:  column "col3" does not exist
LINE 1: DELETE FROM update_delete_char1 WHERE col3 = 'ab';
                                              ^
HINT:  Perhaps you meant to reference the column "update_delete_char1.col1" or the column "update_delete_char1.col2".
SELECT * FROM update_delete_char1 ORDER BY col1;
 col1 | col2 
------+------
    1 | uv
(1 row)

---varchar
SELECT * FROM update_delete_varchar1 ORDER BY col1;
 col1 | col2 
------+------
(0 rows)

INSERT INTO update_delete_varchar1 (col1, col2) VALUES (1, 'ab');
INSERT INTO update_delete_varchar1 (col1, col2) VALUES (2, 'kl');
SELECT * FROM update_delete_varchar1 ORDER BY col1;
 col1 | col2 
------+------
    1 | ab
    2 | kl
(2 rows)

UPDATE update_delete_varchar1 SET col2 = 'uv' WHERE col2 = 'ab';
SELECT * FROM update_delete_varchar1 ORDER BY col1;
 col1 | col2 
------+------
    1 | uv
    2 | kl
(2 rows)

UPDATE update_delete_varchar1 SET col2 = 'uvd' WHERE col2 = 'kl';
ERROR:  Failed to execute statement to Tsurugi. (13)
Tsurugi Server Error: VALUE_TOO_LONG_EXCEPTION (SQL-02058: evaluated value was too long)
SELECT * FROM update_delete_varchar1 ORDER BY col1;
 col1 | col2 
------+------
    1 | uv
    2 | kl
(2 rows)

DELETE FROM update_delete_varchar1 WHERE col2 = 'kl';
SELECT * FROM update_delete_varchar1 ORDER BY col1;
 col1 | col2 
------+------
    1 | uv
(1 row)

DELETE FROM update_delete_varchar1 WHERE col3 = 'ab';
ERROR:  column "col3" does not exist
LINE 1: DELETE FROM update_delete_varchar1 WHERE col3 = 'ab';
                                                 ^
HINT:  Perhaps you meant to reference the column "update_delete_varchar1.col1" or the column "update_delete_varchar1.col2".
SELECT * FROM update_delete_varchar1 ORDER BY col1;
 col1 | col2 
------+------
    1 | uv
(1 row)

/* Test teardown: DDL of the PostgreSQL */
DROP FOREIGN TABLE update_delete_int1;
DROP FOREIGN TABLE update_delete_bigint1;
DROP FOREIGN TABLE update_delete_real1;
DROP FOREIGN TABLE update_delete_double1;
DROP FOREIGN TABLE update_delete_char1;
DROP FOREIGN TABLE update_delete_varchar1;
/* Test teardown: DDL of the Tsurugi */
SELECT tg_execute_ddl('tsurugidb', 'DROP TABLE update_delete_int1');
 tg_execute_ddl 
----------------
 DROP TABLE
(1 row)

SELECT tg_execute_ddl('tsurugidb', 'DROP TABLE update_delete_bigint1');
 tg_execute_ddl 
----------------
 DROP TABLE
(1 row)

SELECT tg_execute_ddl('tsurugidb', 'DROP TABLE update_delete_real1');
 tg_execute_ddl 
----------------
 DROP TABLE
(1 row)

SELECT tg_execute_ddl('tsurugidb', 'DROP TABLE update_delete_double1');
 tg_execute_ddl 
----------------
 DROP TABLE
(1 row)

SELECT tg_execute_ddl('tsurugidb', 'DROP TABLE update_delete_char1');
 tg_execute_ddl 
----------------
 DROP TABLE
(1 row)

SELECT tg_execute_ddl('tsurugidb', 'DROP TABLE update_delete_varchar1');
 tg_execute_ddl 
----------------
 DROP TABLE
(1 row)

